src/lib.rs

```diff
@@ -222,6 +224,68 @@ impl<T: InputLength> InputLength for LocatedSpan<T> {
     }
 }
 
+impl<T: AtEof> AtEof for LocatedSpan<T> {
+    fn at_eof(&self) -> bool {
+        self.fragment.at_eof()
+    }
+}
+
+impl<T> InputTake for LocatedSpan<T> where Self: Slice<RangeFrom<usize>> + Slice<RangeTo<usize>> {
+    fn take(&self, count: usize) -> Self {
+        self.slice(..count)
+    }
+
+    fn take_split(&self, count: usize) -> (Self, Self) {
+        (self.slice(count..), self.slice(..count))
+    }
+}
```

Comment:
Is there any reason for not returning the call of `self.fragment.take` or `self.fragment.take_split`?

Reply:
Maybe i'm wrong, but using `self.fragment.take` or `self.fragment.take_split` would return a value of the fragment's type and not `Self`. Furthermore, `LocatedSpan<T>.offset` and `LocatedSpan<T>.line` are updated by the `slice` method.

Reply:
Ah yeah, obviously, thanks!